src/Peering/Peering/PeeringService/Prefix/GetAzurePeeringServicePrefixCommand.cs

```diff
@@ -0,0 +1,191 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+namespace Microsoft.Azure.PowerShell.Cmdlets.Peering.Peering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+
+    using Microsoft.Azure.Commands.Peering.Properties;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.Management.Peering;
+    using Microsoft.Azure.Management.Peering.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Models;
+
+    /// <inheritdoc />
+    /// <summary>
+    ///     The Get Az InputObject Legacy peering.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzPeeringServicePrefix", SupportsShouldProcess = true, DefaultParameterSetName = Constants.ParameterSetNameByResourceAndName)]
+    [OutputType(typeof(PSPeeringServicePrefix))]
+    public class GetAzurePeeringServicePrefixCommand : PeeringBaseCmdlet
+    {
+        /// <summary>
+        /// Gets or sets the peering service input object
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.PrefixInputObjectHelp,
+            ValueFromPipeline = true,
+            ParameterSetName = Constants.ParameterSetNameDefault)]
+        [ValidateNotNullOrEmpty]
+        public PSPeeringService InputObject { get; set; }
```

Comment:
@rileymckenna you should rename this parameter to be `-PeeringServiceObject`, or just `-PeeringService`

Reply:
@cormacpayne so this was what maddie told me to do. Same with the ShouldProcess = true. So now i am trained. Ill be better.